src/block/heading/style.js

```diff
@@ -0,0 +1,87 @@
+/**
+ * External dependencies
+ */
+import {
+	BlockDiv,
+	Advanced,
+	Typography,
+	Alignment,
+	MarginBottom,
+} from '~stackable/block-components'
+import { getUniqueBlockClass } from '~stackable/util'
+import { useDeviceType, useBlockAttributes } from '~stackable/hooks'
+import { useBlockEditContext } from '@wordpress/block-editor'
+
+/**
+ * WordPress dependencies
+ */
+import { Fragment, renderToString } from '@wordpress/element'
+
+export const HeadingStyles = props => {
+	const {
+		...propsToPass
+	} = props
+
+	const deviceType = useDeviceType()
+	const { clientId } = useBlockEditContext()
+	const attributes = useBlockAttributes( clientId )
+
+	propsToPass.blockUniqueClassName = getUniqueBlockClass( attributes.uniqueId )
+	propsToPass.deviceType = deviceType
+	propsToPass.attributes = { ...attributes, clientId }
+
+	return (
+		<Fragment>
+			<Alignment.Style { ...propsToPass } />
+			<BlockDiv.Style { ...propsToPass } />
+			<Advanced.Style { ...propsToPass } />
+			<Typography.Style { ...{
+				...propsToPass,
+				options: {
+					...propsToPass.options,
+					selector: '.stk-heading__text',
+					hoverSelector: '.stk-heading__text:hover',
```

Comment:
No need for this line. For the most part, we would like the effects to be applied when the block is hovered. There are some exceptions, like the Button Block since it will only look correct if the effect triggers when actually hovering over the button